src/main/java/minegame159/meteorclient/mixin/HandledScreenMixin.java

```diff
@@ -181,7 +181,7 @@ private void drawMapPreview(MatrixStack matrices, ItemStack stack, int x, int y,
         buffer.vertex(x1, y1, z).texture(0.0f, 0.0f).next();
         tessellator.draw();
 
-        MapState mapState = FilledMapItem.getMapState(stack, mc.world);
+        MapState mapState = FilledMapItem.getOrCreateMapState(stack, mc.world);
```

Comment:
Why? Why create the map state if it doesn't exist?

Reply:
I though it might help with the issue, where maps dont render until viewed manually once, but it doesnt seem to do anything different

Reply:
Oh

Reply:
I think it might be just that the server didnt send the data yet because you havent held it yet. If thats the case then it cant really be fixed.

Reply:
Huh. Maybe swapping the map with the selected slot for a single tick might fix that, but that seems like a rather ugly fix